src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java

```diff
@@ -87,6 +92,7 @@ public static Long getNumRows(BigQuery bigQuery, String project, String dataset,
                                                  table, dataset, project));
     }
 
+    LOG.info("Table {} contains {} records", table, bgTable.getNumRows().longValue());
```

Comment:
Please surround with `if (LOG.isInfoEnabled())` `getNumRows` is definitely not free

Reply:
should set bgTable.getNumRows().longValue() in a variable and refer to it here in the log statement before returning it below.

Reply:
The expensive operation here is the `Table bgTable = bigQuery.getTable(tableId)`. In that response, the number of rows is returned.

I will make this a local variable as I'm using this value in 2 places.